src/designSystem/components/Tooltip/Tooltip.tsx

```diff
@@ -161,22 +175,47 @@ export const Tooltip = ({
   orientation,
 }: TooltipProps) => {
   const controlRef = useRef<HTMLDivElement | null>(null);
+  const timeoutRef = useRef<any>(null);
   const [controlRect, setControlRect] = useState<DOMRect | null>(null);
   const [isOpen, setIsOpen] = useState(false);
 
   const internalOnMouseEnter = useCallback(() => {
     if (controlRef.current) {
       const rect = controlRef.current.getBoundingClientRect();
-      setControlRect(rect);
-      setIsOpen(true);
+
+      /**
+       * This forces a commit to hover by the user instead of
+       * firing the tooltip as they idly move their mouse
+       * across the screen. The delay is short enough to not
+       * be too noticeable.
+       */
+      timeoutRef.current = setTimeout(() => {
+        setControlRect(rect);
+        setIsOpen(true);
+      }, 300);
     }
   }, []);
 
   const internalOnMouseLeave = useCallback(() => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+
     setControlRect(null);
     setIsOpen(false);
   }, []);
 
+  useEffect(() => {
+    /**
+     * Force a cleanup of timeouts as the component tears
+     * down. We don't want async code executing after the
+     * component unmounts.
+     */
+    return () => {
```

Comment:
Probably not needed, but I still think its a good practice to keep intervals and timeouts cleared when we tear down a component.